provisioner/provision.js

```diff
@@ -192,9 +194,50 @@ Provisioner.prototype.runAllProvisionersOnce = async function () {
   debug('configured workers:           %j', workerNames);
   debug('managed requests/instances:   %j', this.awsManager.knownWorkerTypes());
 
-  return Promise.all(workerTypes.map(async worker => {
-    return this.provisionType(worker, pricing);
-  }));
+  var forSpawning = [];
+
+  for (var worker of workerTypes) {
+    var change = await this.changeForType(worker, pricing);
+
+    if (change > 0) {
+      debug('%s needs %d capacity created', worker.workerType, change);
+      var bids = worker.determineSpotBids(this.awsManager.ec2.regions, pricing, change);
+      // This could probably be done cleaner
+      for (let bid of bids) {
+        forSpawning.push({workerType: worker, bid: bid});
+      }
+    } else if (change < 0) {
+      var capToKill = -change;
+      debug('%s needs %d capacity destroyed', worker.workerType, capToKill);
+      await this.awsManager.killCapacityOfWorkerType(
+            worker, capToKill, ['pending', 'spotReq']);
+    } else {
+      debug('%s needs no changes', worker.workerType);
+    }
+  }
+
+  var d = delayer(500);
+  var longD = delayer(2000);
+
+  // We want to have a maximum number of attempts
+  var attemptsLeft = forSpawning.length * 2;
```

Comment:
I'm not sure it's a good idea to use attempts across all workerTypes... How about attempts was a property on the `toSpawn` object? Then default to 3 for all workerTypes...

Hmm, maybe, maybe not.... WorkerTypes aren't dependent on single region, so no reason one should fail more than the others...